### PR TITLE
OJ-9902 verbose

### PR DESCRIPTION
**Pull request by alison-j**
Adds some verbose logging (off by default, enabled via config.yml when desired) to Bitbucket Server. New logging is all client-side (not sent).

### CODE REVIEW

File: jf_agent/git/bitbucket_server.py [hunk 1]
```diff
@@ -234,9 +236,11 @@ def _normalize_commit(commit, repo, strip_text_content, redact_names_and_urls):
 
 
 def get_default_branch_commits(
-    client, api_repos, strip_text_content, server_git_instance_info, redact_names_and_urls
+    client, api_repos, strip_text_content, server_git_instance_info, redact_names_and_urls, verbose,
 ):
     for i, api_repo in enumerate(api_repos, start=1):
+        if verbose:
+            print(f"Beginning download of commits for repo {api_repo}")
```
Comment (ametzger) [inline jf_agent/git/bitbucket_server.py:243]: For these to make it to the log file, I believe this should be `agent_logging.log_and_print`

Comment (alison-j) [reply jf_agent/git/bitbucket_server.py:243]: We don't want them in the log file! Verbose logging might grow to include stuff a customer would rather not send us by default (though I'm not sure it does yet)-- this is for them to send us to help debug.

File: jf_agent/config_file_reader.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ class GitConfig:
     git_strip_text_content: bool
     git_redact_names_and_urls: bool
     gitlab_per_page_override: bool
+    git_verbose: bool
```
Comment (mattklein) [inline jf_agent/config_file_reader.py:27]: For completeness we should probably add a mention of this into jf_agent/example.yml

**Review by mattklein — APPROVED** [review]
One comment on modifying example.yml for the new arg

File: jf_agent/git/bitbucket_server.py (hunk 1 above)
Comment (ametzger) [reply jf_agent/git/bitbucket_server.py:243]: Got it, :thumbsup:

**Review by ametzger — APPROVED** [review]

File: jf_agent/config_file_reader.py (hunk 2 above)
Comment (alison-j) [reply jf_agent/config_file_reader.py:27]: Good point!

File: example.yml [hunk 3]
```diff
@@ -180,7 +180,10 @@ git:
   # Redact names and URLs for projects, repos, branches.
   redact_names_and_urls: False
 
-######################################
+  # More verbose local logging (currently only available on Bitbucket Server).
+  verbose: False
+
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
```
Comment (jnicoleoliveira) [inline example.yml:186]: 👀

File: jf_agent/git/bitbucket_server.py [hunk 4]
```diff
@@ -248,13 +252,17 @@ def get_default_branch_commits(
                 default_branch = (
                     api_repo.default_branch['displayId'] if api_repo.default_branch else ''
                 )
+                if verbose:
+                    print("Getting list of commits.")
```
Comment (jnicoleoliveira) [inline jf_agent/git/bitbucket_server.py:256]: I would maybe add the associated repo here: `Getting list of commits for {repo['name']}`

File: jf_agent/git/bitbucket_server.py [hunk 5]
```diff
@@ -289,11 +299,13 @@ def _normalize_pr_repo(repo, redact_names_and_urls):
...
 
 def get_pull_requests(
-    client, api_repos, strip_text_content, server_git_instance_info, redact_names_and_urls
+    client, api_repos, strip_text_content, server_git_instance_info, redact_names_and_urls, verbose
 ):
     for i, api_repo in enumerate(api_repos, start=1):
         with agent_logging.log_loop_iters(logger, 'repo for pull requests', i, 1):
             repo = api_repo.get()
+            if verbose:
+                print(f"Beginning download of PRs for repo {repo}")
```
Comment (jnicoleoliveira) [inline jf_agent/git/bitbucket_server.py:308]: Maybe for consistency, match this with the other verbose terms?  
Apply `Beginning download` to `Getting list of commits` to make `Beginning download of commits for repo {repo}`

File: jf_agent/git/bitbucket_server.py [hunk 6]
```diff
@@ -304,6 +316,8 @@ def get_pull_requests(
                 desc=f'downloading PRs for {repo["name"]}',
                 unit='prs',
             ):
+                if verbose:
+                    print(f"Processing PR {pr['id']}")
```
Comment (jnicoleoliveira) [inline jf_agent/git/bitbucket_server.py:320]: Same here.. maybe do the above to commits.

File: jf_agent/git/bitbucket_server.py [hunk 7]
```diff
@@ -447,3 +461,5 @@ def get_pull_requests(
                 }
 
                 yield normalized_pr
+        if verbose:
+            print(f"Ending download of PRs for repo {repo}")
```
Comment (jnicoleoliveira) [inline jf_agent/git/bitbucket_server.py:465]: I'm not sure how useful start/end logging is here. I think tqdm should be printing this already 🤔

**Review by jnicoleoliveira — COMMENTED** [review]
A few comments on logging verbage consistency

File: example.yml (hunk 3 above)
Comment (alison-j) [reply example.yml:186]: I have no idea what happened there. Fixing.

File: jf_agent/git/bitbucket_server.py (hunk 7 above)
Comment (alison-j) [reply jf_agent/git/bitbucket_server.py:465]: 👍

**Pull request merged**